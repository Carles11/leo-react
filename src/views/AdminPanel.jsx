import React from 'react'

const AdminPanel = () => <article>Admin Panel</article>

export default AdminPanel