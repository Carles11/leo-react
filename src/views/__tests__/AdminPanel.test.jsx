import React from 'react'
import { shallow } from 'enzyme'
import AdminPanel from '../AdminPanel'

const properties = {
  auth: false,
}

describe('<AdminPanel />', () => {
  it('should render itself', () => {
    const component = shallow(<AdminPanel {...properties} />)
    expect(component).toMatchSnapshot()
  })

  it('should redirect to Admin if auth is false', () => {
    const component = shallow(<AdminPanel {...properties} />)
    expect(component.find('Redirect').length).toEqual(1)
  })

  it('should render AdminList if auth is true', () => {
    properties.auth = true
    const component = shallow(<AdminPanel {...properties} />)
    expect(component.find('Redirect').length).toEqual(0)
    expect(component.find('AdminList').length).toEqual(1)
  })
})
