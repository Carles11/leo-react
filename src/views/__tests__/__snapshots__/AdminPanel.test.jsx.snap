// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPanel /> should render itself 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanel
    auth={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Redirect
          push={false}
          to="/admin"
/>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/admin",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Redirect
            push={false}
            to="/admin"
/>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/admin",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
